pkg/client/client.go

```diff
@@ -67,6 +67,9 @@ func (rd *ErrorHandlingClient) Do(req *http.Request) (*http.Response, error) {
 	e := clierr.New(fmt.Sprintf("Common Fate API returned an error (code %v): %s", res.StatusCode, string(body)))
 
 	if res.StatusCode == http.StatusUnauthorized {
+		if cfContext.DashboardURL != "" {
+			return nil, clierr.New(fmt.Sprintf("To log in to Common Fate, run: run: '%s %s'", rd.LoginHint, cfContext.DashboardURL))
+		}
 		e.Messages = append(e.Messages, clierr.Infof("To log in to Common Fate, run: '%s'", rd.LoginHint))
```

Comment:
I think there is a logic error here - returning early will discard the `res` value as you have set it to `nil` in the code you've added.

```suggestion
		if cfContext.DashboardURL != "" {
			e.Messages = append(e.Messages, clierr.Infof("To log in to Common Fate, run: '%s %s'", rd.LoginHint, cfContext.DashboardURL))
		} else {
		        e.Messages = append(e.Messages, clierr.Infof("To log in to Common Fate, run: '%s'", rd.LoginHint))
		 }
```